### DISCUSSION TITLE
Rewards

### DISCUSSION THREAD

**Discussion by APNPAPNU** [dogecoin/dogecoin#2352] [Q&A]
Would it be beneficial to place a cap on mining rewards? Or to burn a certain amount of the dogecoin supply periodically to help counteract inflation?

**Comment by AbcSxyZ** [comment]
Dogecoin is deflationary, no need to counteract inflation :rofl: 

Anyway, mining rewards are used to paid miners, and because fees weren't high and are slowly reduced, it's a critical incentive.
Bitcoin hope transactions fees would pay miners in the future, Dogecoin know how they will be paid in 2140 : using block rewards.

About burning coins, are you agree to burn yours, or only if it's someone else coins ?

For your personal information, in **current** economy theories, deflation for a currency is seen as a threat : it prevents you to spend.  
*Who knows if bitcoin economical paradigm is good...*

The single argument I've seen :

> **Deflationary Money**
> The  most  important  and  debated  consequence  of  a  fixed  and  diminishing  monetary issuance  is  that  the  currency  will  tend  to  be  inherently  deflationary.  Deflation  is  the phenomenon of appreciation of value due to a mismatch in supply and demand that drives up the value (and exchange rate) of a currency. The opposite of inflation, price deflation means that the money has more purchasing power over time.
>
> Many economists argue that a deflationary economy is a disaster that should be avoided at all costs. That is because in a period of rapid deflation, people tend to hoard money instead of spending it, hoping that prices will fall. Such a phenomenon unfolded during Japan’s “Lost Decade,” when a complete collapse of demand pushed the currency into a deflationary spiral.
>
> Bitcoin experts argue that deflation is not bad per se. Rather, deflation is associated with a collapse in demand because that is the only example of deflation we have to study. In a fiat currency with the possibility of unlimited printing, it is very difficult to enter a deflationary spiral unless there is a complete collapse in demand and an unwillingness to  print  money.  Deflation  in  bitcoin  is  not  caused  by  a  collapse  in  demand,  but  by  apredictably constrained supply.
>
> In practice, it has become evident that the hoarding instinct caused by a deflationary currency can be overcome by discounting from vendors, until the discount overcomes the  hoarding  instinct  of  the  buyer.  Because  the  seller  is  also  motivated  to  hoard,  the discount becomes the equilibrium price at which the two hoarding instincts are match‐ed. With discounts of 30% on the bitcoin price, most bitcoin retailers are not experi‐encing difficulty overcoming the hoarding instinct and generating revenue. It remains to be seen whether the deflationary aspect of the currency is really a problem when it is not driven by rapid economic retraction.
>
> *Andreas M. Antonopoulos - Mastering Bitcoin*

We know that we don't know !

**Comment by patricklodder** [comment]
There is a cap on mining rewards, there's just no cap on time. I don't want to make a cap on time because a) I don't know how and b)  I'd like shibes to thrive forever anyway :-)

**Reply by crevolutionX** [reply]
Then who decides if the inflate rate should be 5B coins a year or 10B coins a year or 0.. I think you'd prefer no inflation if you'd hodl but if you'd be making a lot of transactions then you'd want more hashrate to secure the network. Can we come up with a solution that will benefit to both type of users?

> There is a cap on mining rewards, there's just no cap on time. I don't want to make a cap on time because a) I don't know how and b) I'd like shibes to thrive forever anyway :-)

**Reply by AbcSxyZ** [reply]
*[Correct me if I'm wrong, I'm answering with my current understanding of crypto/open source. Related to Dogecoin but not only]*

Important question, but quite complex answer, because blockchain related software are specific type of software, with the **open source** dimension but also because it's a **decentralized network**.  
*Following [Fee Policy proposal](https://github.com/dogecoin/dogecoin/discussions/2347) can be interesting to understand how things works !*

You have to understand the role of this 2 type of actors : **developers** and  **nodes operators**.
- **Developers** produce the software.
- **Nodes operators** run the software.

**Developers** have the skills to improve the software and because it's open source, everyone has access to the code and can suggest change to the repo. In fact, even if it tends to be horizontal there is a centralized aspect in open source, it's managed by a group of people or a single guy, some guys having a specific power : **the committers**. Those guys can accept or reject any change to a main repo, they have to be convinced.
*So anyone [with programming skills] can suggest a change, but it has to be accepted by the committers. If some change is rejected, but you (and a group) want to implement it, a fork may happen.*

**Nodes operators** are important actor because they are the network : they decide to use (or not) the software and his updates provided by developers. They have the power to accept or refuse a change inside the network, they also have to be convinced for any improvement to take effect. Nodes operators and developers are not necessary in a close relation.

For example, imagine you want to increase the block reward to 20'000 coins :
- First you have to convince devs to implement it in the code (through a Dogecoin Improvement Proposal [DIP] I would say, but I think it's actually unclear where it has to be done), and (a majority of/all ?) node operators have to accept to upgrade the software. If not enough node operators upgrade the software (voluntarily or not), the new mined block of 20'000 doge will be considered as invalid by the rest of the network who expect a 10'000 reward.

**To decide a change, it's a political game between multiple actors.**
> Patches that change Bitcoin consensus rules are considerably more involved than normal because they affect the entire ecosystem and so must be preceded by extensive mailing list discussions and have a numbered BIP [Bitcoin Improvement Proposal]. While each case will be different, one should be prepared to expend more time and effort than for other kinds of patches because of increased peer review and consensus building requirements.
> **Bitcoin [contributing guidelines](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#decision-making-process)**

*P.S. : Things can be tricky, see the notion of [bitcoin ossification](https://bengoldhaber.com/posts/bitcoin-ossification) and the [Block Size War](https://keepingstock.net/an-open-letter-to-bitcoin-miners-c260467e1f0).*

**Reply by patricklodder** [reply]
> Can we come up with a solution that will benefit to both type of users?

What makes you think that the current solution is not beneficial for both types of users?
